plugins/spectate.cpp

```diff
@@ -527,7 +530,7 @@ static void follow_a_dwarf(color_ostream &out) {
     intervals.push_back(0);
     add_bucket(citizen_combat_units, units, intervals, weights, config.prefer_conflict ? CITIZEN_COMBAT_PREFERRED_WEIGHT : JOB_WEIGHT);
     add_bucket(other_combat_units, units, intervals, weights, config.prefer_conflict ? OTHER_COMBAT_PREFERRED_WEIGHT : JOB_WEIGHT);
-    add_bucket(nicknamed_units, units, intervals, weights, config.prefer_nicknamed ? NICKNAMED_CITIZEN_PREFERRED_WEIGHT : JOB_WEIGHT);
+    add_bucket(nicknamed_units, units, intervals, weights, config.prefer_nicknamed ? NICKNAMED_CITIZEN_PREFERRED_WEIGHT : OTHER_WEIGHT);
```

Comment:
Actually, sorry, I take this back. If prefer_nicknamed is false, then the units shouldn't even be added to the bucket. That way the job weight will still take effect for the unit.